client/client.go

```diff
@@ -69,19 +78,52 @@ func newClient(cfg *rest.Config) (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(func(format string, args ...interface{}) { kubeapplierlog.Logger.Debug(fmt.Sprintf(format, args...)) })
+	eventBroadcaster.StartRecordingToSink(&clientv1.EventSinkImpl{Interface: clientset.CoreV1().Events("")})
+	recorder := eventBroadcaster.NewRecorder(scheme, corev1.EventSource{Component: clientName})
 	return &Client{
 		Client:    c,
 		clientset: clientset,
+		recorder:  recorder,
 	}, nil
 }
 
+// EmitApplicationEvent creates an Event for the provided Application.
+func (c *Client) EmitApplicationEvent(app *kubeapplierv1alpha1.Application, eventtype, reason, messageFmt string, args ...interface{}) {
+	c.recorder.Eventf(app, eventtype, reason, messageFmt, args...)
+}
+
```

Comment:
```suggestion
func (c *Client) EmitApplicationEvent(app *kubeapplierv1alpha1.Application, eventType, reason, messageFmt string, args ...interface{}) {
	c.recorder.Eventf(app, eventType, reason, messageFmt, args...)
}
```

nit: camel case.

Reply:
I've just copied the args from upstream, https://godoc.org/k8s.io/client-go/tools/record#EventRecorder
I can totally empathise with you)

Reply:
Fair enough 😄